hmmlearn/tests/test_gaussian_hmm.py

```diff
@@ -123,6 +123,7 @@ def test_fit_zero_variance(self):
         h = hmm.GaussianHMM(3, self.covariance_type)
         h.fit(X)
 
+    @pytest.mark.xfail
```

Comment:
I think it works on Mac/Linux. Could you at least mark it as conditional xfail?

Reply:
It definitely fails sporadically on linux, given recent TravisCI builds: https://travis-ci.org/hmmlearn/hmmlearn/jobs/285760409#L588-L641, and I can reproduce it locally on linux by running the test many times. For instance, this loop run on `master` just now yielded 4 failures and 96 successes: `$ for i in {1..100}; do pytest hmmlearn/tests/test_gaussian_hmm.py::TestGaussianHMMWithFullCovars::test_fit_with_priors --disable-warnings; done`. For reference, here is the output: [test_fit_with_priors_x100.txt](https://github.com/hmmlearn/hmmlearn/files/1379548/test_fit_with_priors_x100.txt).

I believe the problem has to do with how the covariance matrix is created using only the global `RandomState` rather than providing it with a `RandomState` that is reproducible here: https://github.com/hmmlearn/hmmlearn/blob/ae1a41e4d03ea61b7a25cba68698e8e2e52880ad/hmmlearn/tests/__init__.py#L10-L23

I was planning on fixing that in a separate PR tonight, but I could include that here if that sounds appropriate to you.

Reply:
For the record, I'm seeing failures here with 0.2.0 on mac.